include/yaclib/coro/detail/await_awaiter.hpp

```diff
@@ -124,4 +124,26 @@ AwaitAwaiter<false>::AwaitAwaiter(It it, std::size_t count) noexcept : _event{co
   _event.count.fetch_sub(count - wait_count, std::memory_order_relaxed);
 }
 
+template<typename V, typename E>
+class [[nodiscard]] AwaitSingleAwaiter final {
+  public:
+  AwaitSingleAwaiter(ResultCorePtr<V, E> result) noexcept : _result{ std::move(result) } {}
+
+  YACLIB_INLINE bool await_ready() const noexcept {
+    return !_result->Empty();
+  }
+
+  template <typename Promise>
+  YACLIB_INLINE bool await_suspend(yaclib_std::coroutine_handle<Promise> handle) noexcept {
+    return _result->SetCallback(handle.promise());
+  }
+
+  constexpr Result<V, E> await_resume() noexcept {
+    return std::move(_result->Get());
+  }
```

Comment:
I think co_await is just wrapper for less typing, so I don't think it's convinient to write `(co_await ...).Ok()`

```suggestion
  V await_resume() {
    return std::move(_result->Get()).Ok();
  }
```

Btw with rust "suffix" `.await` it probably will be ok :(